index.js

```diff
@@ -1,40 +1,31 @@
-module.exports = function(filteredImports) {
-  return function(babel) {
-    // A stack of booleans that determine whether an expression statement
-    // should be removed as it is exited. Expression statements are removed
-    // when they contain a reference to a filtered imported.
-    var shouldRemove;
-
-    return new babel.Transformer('babel-plugin-filter-imports', {
-      Program: {
-        enter: function() {
-          shouldRemove = [];
-        },
-        exit: function() {
-          shouldRemove = undefined;
-        }
-      },
```

Comment:
Funny enough this actually worked without changes even though the instances shared the array. I've changed it to one array per instance now since it's a much cleaner implementation.